src/Compilers/CSharp/Test/Syntax/Parsing/LocalFunctionParsingTests.cs

```diff
@@ -607,5 +607,277 @@ static void H()
+                            N(SyntaxKind.Block);
+                            {
+                                N(SyntaxKind.OpenBraceToken);
+                                N(SyntaxKind.LocalFunctionStatement);
+                                {
+                                    N(SyntaxKind.StaticKeyword);
+                                    N(SyntaxKind.AsyncKeyword);
+                                    N(SyntaxKind.PredefinedType);
+                                    N(SyntaxKind.VoidKeyword);
+                                    N(SyntaxKind.IdentifierToken, "F1");
+                                    N(SyntaxKind.ParameterList);
+                                    {
+                                        N(SyntaxKind.OpenParenToken);
+                                        N(SyntaxKind.CloseParenToken);
+                                    }
+                                    N(SyntaxKind.Block);
+                                    {
+                                        N(SyntaxKind.OpenBraceToken);
+                                        N(SyntaxKind.CloseBraceToken);
+                                    }
+                                }
+                                N(SyntaxKind.LocalFunctionStatement);
+                                {
+                                    N(SyntaxKind.AsyncKeyword);
+                                    N(SyntaxKind.StaticKeyword);
+                                    N(SyntaxKind.PredefinedType);
+                                    N(SyntaxKind.VoidKeyword);
+                                    N(SyntaxKind.IdentifierToken, "F2");
+                                    N(SyntaxKind.ParameterList);
+                                    {
+                                        N(SyntaxKind.OpenParenToken);
+                                        N(SyntaxKind.CloseParenToken);
+                                    }
+                                    N(SyntaxKind.Block);
+                                    {
+                                        N(SyntaxKind.OpenBraceToken);
+                                        N(SyntaxKind.CloseBraceToken);
+                                    }
+                                }
+                                N(SyntaxKind.CloseBraceToken);
+                            }
+                        }
+                    }
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                EOF();
+            }
+        }
+
+        [Fact]
+        public void DuplicateStatic()
+        {
+            const string text =
+@"class Program
+{
+    void M()
+    {
+        static static void F1() { }
+        static async static void F2() { }
+    }
+}";
+
+            UsingDeclaration(text, options: TestOptions.Regular7_3,
+                // (5,9): error CS8421: To use the 'static' modifier for local functions, please use language version 8.0 or greater.
+                //         static static void F1() { }
```

Comment:
>static static [](start = 27, length = 13)

nit: the error recovery seems worse than on regular methods. I'd expect a single diagnostic: "error CS1004: Duplicate 'static' modifier".
Consider filing an issue.

[Example](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLRIMaOQSwG4SoAOsMECAdsgD4ACADAAS0CMA3ALABQtAzMwCZGAYUYBvboynN+rAGzMWC2gBZGAWQAUASnGTpAX24GgA=)

Reply:
Logged [https://github.com/dotnet/roslyn/issues/32106](https://github.com/dotnet/roslyn/issues/32106)

---
In reply to: [244819913](https://github.com/dotnet/roslyn/pull/32067#discussion_r244819913) [](ancestors = 244819913)